### PR TITLE
[#41] Code assist should not put unnecessary quotes around string value

### PR DESCRIPTION
**Pull request by tedepstein**
Remove unnecessary implementations of ICompletionProposals and use ContentAssist options to replicate their behaviours.

Re-creating the Pull Request that I prematurely merged, and then reverted, late last night.

### CODE REVIEW

**Comment by tedepstein** [comment]
@ghillairet , you had opened PR #63 for this merge.  Last night I accidentally merged it, then reverted.  This new Pull Request replaces #63, but apparently there are merge conflicts that need to be resolved.

Also, Tanya noted some questions about the code, and a test failure.  Please see her comments in #63.

**Comment by ghillairet** [comment]
Fixed conflicts.

File: com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerSourceViewerConfiguration.java [hunk 1]
```diff
@@ -25,10 +25,11 @@ public IContentAssistant getContentAssistant(ISourceViewer sourceViewer) {
 		ca.setContentAssistProcessor(new SwaggerContentAssistProcessor(), IDocument.DEFAULT_CONTENT_TYPE);
 		ca.setInformationControlCreator(getInformationControlCreator(sourceViewer));
 
-		ca.enableAutoInsert(true);
+		ca.enableAutoInsert(false);
```
Comment (tfesenko) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/editor/SwaggerSourceViewerConfiguration.java:28]: Thanks

File: com.reprezen.swagedit/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessor.java [hunk 2]
```diff
@@ -77,12 +78,28 @@ public SwaggerContentAssistProcessor() {}
...
+		if (i > document.getLength())
+			return ""; //$NON-NLS-1$
 
-		return result;
+		try {
+			while (i > 0) {
+				char ch= document.getChar(i - 1);
+				if (Character.isWhitespace(ch))
+					break;
+				i--;
```
Comment (tfesenko) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessor.java:96]: I don't understand what exactly this method does. The difference with the super implementation is that it uses `!Character.isJavaIdentifierPart(ch)` and we use `Character.isWhitespace(ch)`, but what's its purpose? A comment would be helpful here.

Comment (ghillairet) [inline com.reprezen.swagedit/src/com/reprezen/swagedit/assist/SwaggerContentAssistProcessor.java:96]: It's to get the quotes, the method isJavaIdentifierPart does not include them.
It's here to support the case MG describes in https://github.com/ModelSolv/SwagEdit/issues/42

**Comment by tfesenko** [comment]
Looks good. Merging. I left a comment in https://github.com/ModelSolv/SwagEdit/issues/42#issuecomment-171710372. 
But I would also appreciate a comment https://github.com/ModelSolv/SwagEdit/pull/78#discussion_r49752357 directly in the code.

**Pull request merged**